### DISCUSSION TITLE
Migrated to s3 - slashes in identifiers

### DISCUSSION THREAD

**Discussion by sotiriosa-cogapp** [cantaloupe-project/cantaloupe#588] [Q&A]
I have migrated an image server (not Cantaloupe) from a local server to Cantaloupe on AWS. I have moved the images and maintained the existing file structure in s3, so some of the images are in subdirectories within the bucket that Cantaloupe is serving images from.

However, those images that are within sub-directories can no longer be accessed from the same URL structure as before. Here's an example to illustrate what I mean:
Image id: `subfolder/image.tif`
Old url: `example-old-url.com/iiif/2/subfolder/image.tif/full/full/0/default.jpg`
New url (not working): `example-new-url.com/iiif/2/subfolder/image.tif/full/full/0/default.jpg`
New url (working): `example-new-url.com/iiif/2/subfolder%2image.tif/full/full/0/default.jpg`

I need to be able to preserve the existing URLs, and it seems to me that I need the reverse of the `slash_substitute` config key. Can I achieve what I want using a ScriptLookupStrategy?
Any help or suggestions greatly appreciated. 
Edit: included the IIIF api number in my example URLs (thanks @adolski )

**Comment by adolski** [comment]
Hi @sotiriosa-cogapp ,

All of Cantaloupe's IIIF Image API endpoints have a path ending in the endpoint's Image API version number:

* `/iiif/1`
* `/iiif/2`
* `/iiif/3`

These are incompatible with that of your previous image server (`/iiif`).

The other problem is that Cantaloupe (per the IIIF Image API) does not allow unencoded slashes in the identifier portion of an Image API URL. Slashes must be percent-encoded, or else the `slash_substitute` config key used to substitute a different character sequence for the slash. But either way would not be what you need.

So, to preserve your URLs, you would have to access Cantaloupe through a reverse proxy using e.g. mod_rewrite or equivalent, with it set up to add a version number to the endpoint path, and escape slashes in the identifier portion of the URL.

**Reply by sotiriosa-cogapp** [reply]
Thanks @adolski , I had omitted the IIIF API version number from my examples through haste. Thank you for the super useful suggestion, I will give that a whirl.
